ffi/capi/c/examples/fixeddecimal_tiny/Makefile

```diff
@@ -27,13 +27,13 @@ icu_capi_staticlib_tiny/target/x86_64-unknown-linux-gnu/debug/libicu_capi_static
 	rustup toolchain install ${ICU4X_NIGHTLY_TOOLCHAIN}
 	rustup component add rust-src --toolchain ${ICU4X_NIGHTLY_TOOLCHAIN}
 	cd icu_capi_staticlib_tiny && \
-	RUSTFLAGS="-Clinker-plugin-lto -Clinker=$(CLANG) -Ccodegen-units=1 -Clink-arg=-flto -Cpanic=abort" ICU4X_DATA_DIR=$(shell pwd)/baked_data cargo +${ICU4X_NIGHTLY_TOOLCHAIN} panic-abort-build --target x86_64-unknown-linux-gnu
+	RUSTFLAGS="-Clinker-plugin-lto -Clinker=$(CLANG) -Ccodegen-units=1 -Clink-arg=-flto -Cpanic=abort" ICU4X_DATA_DIR=$(shell pwd)/baked_data cargo +${ICU4X_NIGHTLY_TOOLCHAIN} build -Z build-std=std,panic_abort -Z build-std-features=panic_immediate_abort --target x86_64-unknown-linux-gnu
 
-icu_capi_staticlib_tiny/target/x86_64-unknown-linux-gnu/release-opt-size/libicu_capi_staticlib_tiny.a: FORCE baked_data/macros.rs
+icu_capi_staticlib_tiny/target/x86_64-unknown-linux-gnu/release/libicu_capi_staticlib_tiny.a: FORCE baked_data/macros.rs
 	rustup toolchain install ${ICU4X_NIGHTLY_TOOLCHAIN}
 	rustup component add rust-src --toolchain ${ICU4X_NIGHTLY_TOOLCHAIN}
 	cd icu_capi_staticlib_tiny && \
-	RUSTFLAGS="-Clinker-plugin-lto -Clinker=$(CLANG) -Ccodegen-units=1 -Clink-arg=-flto -Cpanic=abort" ICU4X_DATA_DIR=$(shell pwd)/baked_data cargo +${ICU4X_NIGHTLY_TOOLCHAIN} panic-abort-build --target x86_64-unknown-linux-gnu --profile=release-opt-size
+	RUSTFLAGS="-Clto -Cembed-bitcode -Clinker-plugin-lto -Clinker=$(CLANG) -Ccodegen-units=1 -Clink-arg=-flto -Cpanic=abort -Copt-level=s" ICU4X_DATA_DIR=$(shell pwd)/baked_data cargo +${ICU4X_NIGHTLY_TOOLCHAIN} build -Z build-std=std,panic_abort -Z build-std-features=panic_immediate_abort --target x86_64-unknown-linux-gnu --release
```

Comment:
My preferred outcome would be one where as many of these settings as possible are in the Cargo.toml file and not `.cargo/config` or floating CLI arguments.

Let's look at each of them:

- `-Clto -Cembed-bitcode -Clinker-plugin-lto` seem to all be controllable by Cargo.toml `lto`
- `-Clinker=$(CLANG)` might need to stay here since we need clang and it is a variable
- `-Ccodegen-units=1` is in Cargo.toml as `codegen-units`
- `-Clink-arg=-flto` seems surprising if it is actually necessary. I think enabling LTO in Cargo.toml should be enough; we shouldn't need to re-enable it on the linker again. But, maybe setting a custom linker means we need to pass this option back in again.
- `-Cpanic=abort` is in Cargo.toml as `panic = "abort"`
- `-Copt-level=s` is in Cargo.toml as `opt-level = "s"`
- `-Z build-std=std,panic_abort -Z build-std-features=panic_immediate_abort` is unstable so it appears it needs to go either here or in .cargo/config unfortunately

Reply:
I don't think "as many of these settings as possible" in Cargo.toml is desirable. It would be nice if *all* of them could be in Cargo.toml, but that's not possible right now. Having them split across two files is confusing, because it's unclear how they interact, *and* they're split across two files.

Reply:
I still think we should use Cargo profiles when available but I won't die on this hill because the PR has other improvements over the status quo